### ISSUE TITLE
Crash when getting geo position

### ISSUE THREAD

**Issue by lupacjiri** [gavineadie/SatelliteKit#2]
### Summary

App crashes when `geoPosition(julianDays:)` is called.

### STR

1. use [this](https://celestrak.com/NORAD/elements/starlink.txt) data set
2. parse data using:
```
let satellites = preProcessTLEs(content)
             .filter { formatOK($0.1, $0.2) }
             .compactMap { try? TLE($0.0, $0.1, $0.2) }
             .map { Satellite(withTLE: $0) }
```
3. try to get geo location for each satellite using:
```
let date = Date().julianDate
satellites.forEach { sat in
     let lat = sat.geoPosition(julianDays: date).lat
     let lon = sat.geoPosition(julianDays: date).lon
}
```
4. app crashes on `SatelliteKit/Satellite.swift:72: Fatal error: Satellite Position Error FALCON 9 DEB .. SatelliteKit.SatKitError (1)`

Is this a bug, or am I doing something wrong? It works just fine for some satellites in a set, but crashes on FALCON 9 DEB each time.

**Comment by wolfcon** [comment]
I tried your code, and all fine. No crash.🧐

```swift
    func testIssue2() {
        let contents = try! String(contentsOfFile: "/<#path#>/starlink.txt")
        let satellites = preProcessTLEs(contents)
                     .filter { formatOK($0.1, $0.2) }
                     .compactMap { try? TLE($0.0, $0.1, $0.2) }
                     .map { Satellite(withTLE: $0) }
        let date = Date().julianDate
        satellites.forEach { sat in
             let lat = sat.geoPosition(julianDays: date).lat
             let lon = sat.geoPosition(julianDays: date).lon
        }
    }
```
[starlink.txt.zip](https://github.com/gavineadie/SatelliteKit/files/7081923/starlink.txt.zip)

**Comment by gavineadie** [comment]
Is this resolved?  Just checking ..

**Comment by s-surjeet** [comment]
Reproducible with below  info:

 "TLE_LINE0": "0 STARLINK-30"
 "TLE_LINE1": "1 44244U 19029K   20287.12291165  .47180237  12426-4  22139-2 0  9995"
 "TLE_LINE2": "2 44244  52.9708 332.0356 0003711 120.7278 242.0157 16.43170483 77756"

**Comment by gavineadie** [comment]
You are correct that this fails; thanks for the test case TLEs.  Interestingly, it propagates successfully to 10 hours after the epoch, but fails before 11 hours.  During that interval it descends below 50 kilometers and is doomed.  I could provide a better error message .. and will.

**Comment by schrockwell** [comment]
I am running into this as well, when calling `Satellite.position`:

```
SatelliteKit/Satellite.swift:108: Fatal error: Position for 'YUSAT 1' .. ERROR 6: decay condition .. radius < 1.0 
```

Sure enough, it's no longer in orbit. As in, it just burned up within the past ten minutes.

The problem is that `position` catches `SatKitError.SGP` and then calls `fatalError`, which is unrecoverable by callers into the library.

I will work on a PR.

**Comment by schrockwell** [comment]
If anyone needs a quick fix, here is a fork of v1.2.1 that adds `throws` to all the prediction methods: https://github.com/schrockwell/SatelliteKit/tree/allow-throws

This is just a independent, temporary fix. I wouldn't expect it to match the final API of the next library version.

**Comment by gavineadie** [comment]
This is tricky to "fix" without breaking existing code so what I've done is add throwing functions with the same names appended with `_throwz` as a temporary measure.  This is ugly and required some other adjustments but it will suffice till the next MAJOR release, when the non-throwing versions will be removed.
